ablySpec/Auth.swift

```diff
@@ -529,5 +529,256 @@ class Auth : QuickSpec {
+                let options = AblyTests.commonAppSetup()
+
+                // ClientId null
+                waitUntil(timeout: testTimeout) { done in
+                    ARTRest(options: options).auth.authorise(nil, options: nil) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expect(tokenDetails).toNot(beNil())
+                        expect(tokenDetails?.clientId).to(beNil())
+                        done()
+                    }
+                }
+
+                options.clientId = "client_string"
+
+                // ClientId not null
+                waitUntil(timeout: testTimeout) { done in
+                    ARTRest(options: options).auth.authorise(nil, options: nil) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expect(tokenDetails).toNot(beNil())
+                        expect(tokenDetails?.clientId).to(equal(options.clientId))
+                        done()
+                    }
+                }
+            }
+
+            // RSA10i
+            context("should adhere to all requirements relating to") {
+
+                it("TokenParams and authCallback") {
+                    let options = AblyTests.commonAppSetup()
+                    options.clientId = "client_string"
+                    let rest = ARTRest(options: options)
+
+                    let expectedClientId = "client_from_callback"
+
+                    let tokenParams = ARTAuthTokenParams()
+                    tokenParams.clientId = "client_from_params"
+
+                    let authOptions = ARTAuthOptions()
+                    authOptions.authCallback = { tokenParams, callback in
+                        rest.auth.createTokenRequest(ARTAuthTokenParams(clientId: expectedClientId), options: options, callback: callback)
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.authorise(tokenParams, options: authOptions) { tokenDetails, error in
+                            expect(error).to(beNil())
+                            expectTokenDetails(tokenDetails)
+                            expect(tokenDetails?.clientId).to(equal(expectedClientId))
+                            done()
+                        }
+                    }
+                }
+
+                it("authUrl") {
+                    let options = AblyTests.commonAppSetup()
+                    options.clientId = "client_string"
+                    let rest = ARTRest(options: options)
+
+                    let authOptions = ARTAuthOptions()
+                    authOptions.authUrl = NSURL(string: "http://auth.ably.io")!
+
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.authorise(nil, options: authOptions) { tokenDetails, error in
+                            expect(error).toNot(beNil())
+                            expect(tokenDetails).to(beNil())
```

Comment:
Ok, makes sense.  Out of interest, how will we ever test that URL auth works? You may want to implement something like https://github.com/ably/ably-js/blob/334c260e192ae417f5b679013cdd6a62cf804c44/spec/realtime/auth.test.js#L53-L107 as well, one for a text token and another for a TokenDetails JSON object